tools/hotspots_annotation.cwl

```diff
@@ -0,0 +1,30 @@
+cwlVersion: v1.0
+class: CommandLineTool
+id: hotspots_annotation
+requirements:
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+  - class: ResourceRequirement
+    ramMin: ${ return inputs.ram * 1000 }
+    coresMin: $(inputs.cores)
+  - class: DockerRequirement
+    dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/hotspots:0.1.0'
+
+baseCommand: []
+
+arguments:
+  - position: 0
+    shellQuote: false
+    valueFrom: >-
+      $(inputs.disable_hotspot_annotation ? ">&2 echo 'User elected to skip hotspot annotation' && exit 0;" : ">&2 /hotspot.py")
+
+inputs:
+  disable_hotspot_annotation: { type: 'boolean?', doc: "Disable Hotspot Annotation and skip this task." }
+  input_vcf: { type: 'File', inputBinding: { position: 99, prefix: '--vcf', shellQuote: false }, secondaryFiles: [.tbi], doc: "VCF file to annotate hotspots." }
+  genomic_hotspots: { type: 'File[]?', inputBinding: { position: 1, prefix: '--genomic_hotspots', shellQuote: false }, doc: "Tab-delimited BED formatted file(s) containing hg38 genomic positions corresponding to hotspots" }
+  protein_hotspots: { type: 'File[]?', inputBinding: { position: 1, prefix: '--protein_hotspots', shellQuote: false }, doc: "Column-name-containing, tab-delimited file(s) containing protein names and HGVSp short values corresponding to hotspots" }
+  output_basename: { type: 'string?', inputBinding: { position: 10, prefix: '--output_basename', shellQuote: false }, doc: "String to use as basename for output file" }
```

Comment:
Could I ask about these inputBinding positions? The two hotspots inputs are both 1, the output is 10, and the input vcf is 99. What is the effect of doing it this way?

Reply:
It has no functional purpose. I just did it for cosmetic reasons. I always want the input VCF to be last and the hotspots files first. As for why output_basename is 10 instead of 2: if I ever wanted to add more bindings between the basename and hotspots files, it gives me more room to do so.